corehq/apps/users/static/users/js/roles.js

```diff
@@ -67,6 +67,10 @@ hqDefine('users/js/roles',[
                 self.specific().forEach(item => item.value(false));
             }
         });
+
+        self.hasError = ko.pureComputed(() => {
+            return self.selection() === selected && permissionModel.filteredSpecific().length == 0;
```

Comment:
Nitpick: uppercasing `selected` would make it more obvious it's a constant

Reply:
b7f3bc38675